### PR TITLE
Make guzzle retry 5xx and 429 responses

### PR DESCRIPTION
**Pull request by GrahamCampbell**

### CODE REVIEW

File: src/Client.php [hunk 1]
```diff
@@ -51,12 +57,30 @@ public function __construct(string $accessToken, GuzzleClient $client = null, in
...
+     * @return \GuzzleHttp\HandlerStack
+     */
+    protected static function createHandler()
+    {
+        $stack = HandlerStack::create();
+
+        $stack->push(Middleware::retry(function ($retries, $request, $response = null, $exception = null) {
+            return $retries < 3 && ($exception instanceof ConnectException || ($response && ($response->getStatusCode() >= 500 || in_array($response->getStatusCode(), self::RETRY_CODES, true))));
+        }, function ($retries) {
+            return (int) pow(2, $retries) * self::RETRY_BACKOFF);
```
Comment (noplanman) [inline src/Client.php]: Missing grouping bracket:
```php
(int) pow(...
```
should be
```php
(int) (pow(...
```

Comment (GrahamCampbell) [reply src/Client.php]: The bracket is not missing. It's not needed. Only the output of pow needs casting.

Comment (GrahamCampbell) [reply src/Client.php]: Oh, there's not meant to be a bracket there at all actually. ;)

**Comment by freekmurze** [comment]
Thanks!

**Pull request merged**